spec/lib/validation_spec.rb

```diff
@@ -360,14 +360,36 @@
   end
 
   describe "#validate_vcpu_quota" do
+    let(:project) { Project.create(name: "test-quota-project") }
+
+    before do
+      project.add_quota(quota_id: ProjectQuota.default_quotas["VmVCpu"]["id"], value: 10)
+    end
+
     it "sufficient cpu quota" do
-      p = instance_double(Project, current_resource_usage: 5, effective_quota_value: 10, quota_available?: true)
-      expect { described_class.validate_vcpu_quota(p, "VmVCpu", 2) }.not_to raise_error
+      # In total, one VM occupying five plus a request for two more,
+      # seven total, is less than the fixture total of ten.
+      Vm.create_with_id(
+        Vm.generate_uuid,
```

Comment:
No need to use `create_with_id` if you are going to pass it `generate_uuid`:

```suggestion
      Vm.create(
```